lib/dialects/abstract/query-generator.js

```diff
@@ -2240,6 +2240,11 @@ var QueryGenerator = {
 
       if (escapeValue) {
         value = this.escape(value, field, escapeOptions);
+
+        //if ANY is used with like, add parentheses to generate correct query
+        if (escapeOptions.acceptStrings && (comparator.indexOf('ANY') > comparator.indexOf('LIKE'))) {
+         value = '(' + value + ')';
+        }
```

Comment:
indexOf check on an array is more precise than on a string though :)

We'll move forward with this fix for now.